app/assets/javascripts/specifics.js.coffee

```diff
@@ -564,7 +590,23 @@ class Row
   toHashKey: ->
     @flattenToIds().join(",") + ",#{@specificLeftMost}" + ",#{@nonSpecificLeftMost?.id}"
 
-  
+  # If the row references a leftmost, either specific or not, return the event(s)
+  # (because a UNION can place multiple events in the specific leftMost, this can be > 1)
+  leftMostEvents: ->
+    if @nonSpecificLeftMost?
+      return [@nonSpecificLeftMost]
+    if @specificLeftMost? && _.isString(@specificLeftMost)
+      specificIndex = hqmf.SpecificsManager.getColumnIndex(@specificLeftMost)
+      return [@values[specificIndex]] if @values[specificIndex]?
+    if @specificLeftMost? && _.isObject(@specificLeftMost)
+      events = []
+      for id, occurrences of @specificLeftMost
+        for occurrence in occurrences
+          specificIndex = hqmf.SpecificsManager.getColumnIndex(occurrence)
+          events.push(@values[specificIndex]) if @values[specificIndex]?
+      return events
+    return []
+
```

Comment:
Good point, I was staring at the extractEventsForLeftMost code and realizing I couldn't use it as is, didn't consider refactoring that.